src/send/test.rs

```diff
@@ -0,0 +1,222 @@
+use rocket::{
+  self,
+  http::{ContentType, Status},
+  local::Client,
+};
+
+fn setup() -> Client
+{
+  let server = rocket::ignite().mount("/", routes![super::handler]);
+
+  Client::new(server).unwrap()
+}
+
+#[test]
+fn single_recipient()
+{
+  let client = setup();
+
+  let mut response = client
+    .post("/send")
+    .header(ContentType::JSON)
+    .body(
+      r#"{
+      "to": [ "foo@example.com" ],
+      "cc": [],
+      "subject": "bar",
+      "body": {
+        "text": "baz",
+        "html": "<a>qux</a>"
+      }
+    }"#,
+    )
+    .dispatch();
+
+  assert_eq!(response.status(), Status::Ok);
+
+  let body = json!(response.body().unwrap().into_string().unwrap());
+  assert_eq!(body, "{\"messageId\":0}");
+}
+
+#[test]
+fn multiple_recipients()
+{
+  let client = setup();
+
+  let mut response = client
+    .post("/send")
+    .header(ContentType::JSON)
+    .body(
+      r#"{
+      "to": [ "foo@example.com", "bar@example.com" ],
+      "cc": [ "baz@example.com", "qux@example.com" ],
+      "subject": "wibble",
+      "body": {
+        "text": "blee",
+        "html": ""
+      }
+    }"#,
+    )
+    .dispatch();
+
+  assert_eq!(response.status(), Status::Ok);
+
+  let body = json!(response.body().unwrap().into_string().unwrap());
+  assert_eq!(body, "{\"messageId\":0}");
+}
+
+#[test]
+fn without_optional_data()
+{
+  let client = setup();
+
+  let mut response = client
+    .post("/send")
+    .header(ContentType::JSON)
+    .body(
+      r#"{
+      "to": [ "foo@example.com" ],
+      "subject": "bar",
+      "body": {
+        "text": "baz"
+      }
+    }"#,
+    )
+    .dispatch();
+
+  assert_eq!(response.status(), Status::Ok);
+
+  let body = json!(response.body().unwrap().into_string().unwrap());
```

Comment:
probably don't need json!() here? Handy for the following line, though:

` assert_eq(body, json!({"messageId": 0}).to_string());`